include/gridtools/common/for_each.hpp

```diff
@@ -53,8 +54,11 @@ namespace gridtools {
 #endif
         } // namespace for_each_impl_
 
-        template <class L>
-        constexpr for_each_impl_::for_each_f<L> for_each = {};
+        template <typename L, typename Args>
+        GT_TARGET GT_FORCE_INLINE GT_TARGET_CONSTEXPR void for_each(Args&& args) {
+            for_each_impl_::for_each_f<L>()(std::forward<Args&&>(args));
```

Comment:
please use `wstd::forward`
double ampersand is wrong here.

Reply:
sure, I did this a bit too much in a rush, thanks! For the workaround here, it seems like the first implementation is incorrect. Relaxed constexpr says that one can skip `__device__` for constexpr functions, but it does not say it can be skipped for constexpr variable templates. So the actual fix that should work without a ifdef is to add `__device__` to the constexpr variable template. Is this pattern also used in other places in the code?

Reply:
yes. this is used in `tuple_util.hpp`. See `DEFINE_TEMPLATED_FUNCTOR_INSTANCE` macro

Reply:
ok, so there it looks good then. I reverted this workaround and added `GT_TARGET` instead. This might have an additional `__host__`, but in my eyes it should be sufficient. What do you think?